blog/2023-11-angular17/README.md

```diff
@@ -0,0 +1,427 @@
+Probieren Sie doch zum Beispiel einmal den neuen Control Flow aus!
+
+
+## Unterstützte Versionen von TypeScript und Node.js
+
+Um Angular 17 zu nutzen, sind die folgenden Versionen von TypeScript und Node.js notwendig:
+
+- **TypeScript 5.2**. Der Support für TypeScript-Versionen kleiner als 5.2 wurde eingestellt.
+- **Node.js 18.13.0**. Node.js in Version 16 wird nicht mehr unterstützt.
+
+
+
+
+
+## Signals gelten als stable
+
+Mit Angular 16 wurde das Konzept der Signals in Angular eingeführt. 
+Im Blogpost *[Angular 16 ist da!](/blog/2023-05-angular16)* haben wir die Ideen dieser neuen *Reactive Primitive* genauer beschrieben.
+Die Signals-Bibliothek von Angular gilt nun als *stable*, sodass Signals auch in produktiven Anwendungen genutzt werden können.
+
+Die Reise mit Signals ist damit aber noch nicht vorbei: In den nächsten Monaten werden weitere Neuerungen kommen, vor allem im Blick auf vollständig signal-basierte Komponenten ([siehe RFC auf GitHub](https://github.com/angular/angular/discussions/49682)).
+Damit wird es möglich sein, Angular-Anwendungen "zoneless" zu betreiben, also ohne die Bibliothek zone.js.
+Dieses Hilfsmittel ist seit jeher notwendig, um die Change Detection von Angular zu triggern – der Prozess, der Änderungen an den Daten ermittelt und die Views der Anwendung automatisch aktualisiert.
+Um diesen Ablauf gezielter und performanter durchzuführen, sind Signals ein wichtiger Grundbaustein.
+
+Übrigens: Die Signal-Methode `mutate()` ist in diesem Release nicht mehr enthalten.
+Die Hintergründe sind im zugehörigen [Commit auf GitHub](https://github.com/angular/angular/commit/c7ff9dff2c14aba70e92b9e216a2d4d97d6ef71e) ausgeführt.
+Statt `mutate()` muss also die Methode `update()` verwendet werden. Dabei müssen Arrays und Objekt als *immutable* behandelt werden.
+
+
+
+
+## Neuer Control Flow: `@if`, `@for`, `@switch`
+
+Die bekannten Strukturdirektiven `NgIf`, `NgFor` und `NgSwitch` sind in ihrer Funktionsweise stark abhängig von zone.js.
+Auf dem Weg zu "Zoneless Angular Apps" musste das Angular-Team das Konzept dieser Direktiven überdenken.
+
+Mit Angular 17 ist nun der neue *Control Flow* für Templates verfügbar!
+Damit wird die Funktionalität der bekannten Direktiven direkt in die Template-Syntax integriert.
+Diese Neuerung hat auch bei der Entwicklung einen entscheidenen Vorteil: Die Syntax wird vom Compiler ausgewertet, und es ist nicht mehr notwendig, die Direktiven bei Standalone Components zu importieren, damit sie überhaupt in der Komponente genutzt werden können.
+
+Die Ausdrücke für den neuen Control Flow werden direkt im HTML-Code notiert und mit einem `@`-Symbol eingeleitet.
+
+
+### Bedingungen mit dem `@if`-Block
+
+Der `@if`-Block dient dazu, bestimmte Teile der Benutzeroberfläche nur dann anzuzeigen, wenn eine Bedingung erfüllt ist. Er ersetzt die Direktive `*ngIf`.
```

Comment:
Ich störe mich an "Benutzeroberfläche". Das ist ein logisches Konzept beim Design, aber kein Teil des Frameworks. Angular kennt Templates und Views.

Reply:
Ich wollte es absichtlich simpel halten

Reply:
Was spricht gegen "Template"? Dass das HTML eienr Komponente so genannt wird, sollte doch bekannt sein.

Reply:
```suggestion
Der `@if`-Block dient dazu, bestimmte Teile des Templates nur dann anzuzeigen, wenn eine Bedingung erfüllt ist. Er ersetzt die Direktive `*ngIf`.
```

Reply:
hab ich schon in meinem Commit geändert

Reply:
Spricht nichts dagegen, aber du hast was dazu geschrieben und ich dachte, du willst ne Antwort! 😄